feat(docs): how to guide for sqlmesh (#1943)
kestra-io/docs: content/docs/15.how-to-guides/sqlmesh.md

---

> title: Using SQLMesh

Reviewer: We can put the full title in here now we have the new how-to guide search page
```suggestion
title: Using SQLMesh to run dbt Projects
```

icon: /docs/icons/tutorial.svg
stage: Getting Started
topics:
  - Integrations
version: ">= 0.18.0"
---

> Using SQLMesh to run dbt project via Kestra

Reviewer: ```suggestion
Using SQLMesh to run dbt project with Kestra.
```

> ## Introduction to SQLMesh

Reviewer: ```suggestion
## What is SQLMesh?
```

> SQLMesh is an open source data transformation and modelling framework. It is a Python framework that automates everything needed to run a scalable data transformation platform. SQLMesh works with a variety of [engines and orchestrators](https://sqlmesh.readthedocs.io/en/stable/integrations/overview/).

Reviewer: ```suggestion
SQLMesh is an open source python data transformation and modelling framework. It automates everything needed to run a scalable data transformation platform. SQLMesh works with a variety of [engines and orchestrators](https://sqlmesh.readthedocs.io/en/stable/integrations/overview/).
```

SQLMesh enables data teams to efficiently run and deploy data transformations written in SQL or Python.

> In this guide, we will learn how to run dbt project based on BigQuery using SQLMesh via Kestra.

Reviewer: ```suggestion
In this guide, we will learn how to run dbt projects based on BigQuery using SQLMesh with Kestra.
```

> ## SQLMesh based flow

Reviewer: ```suggestion
## Example
```

> 

Reviewer: ```suggestion
```

> ### Introduction to Flow

Reviewer: ```suggestion
```

> Here is what we are going to do in the Kestra flow:

Reviewer: ```suggestion
Our Flow will do the following steps:
```

> 1. Download the `orders.csv` using http download task.

Reviewer: ```suggestion
1. Download `orders.csv` using HTTP download task.
```

2. Create the table in BigQuery.

> 3. Upload the data from the csv file into BigQuery table.

Reviewer: ```suggestion
3. Upload the data from the csv file into the BigQuery table.
```

> 4. Create dbt project which will create the BigQuery view from the BigQuery table.

Reviewer: ```suggestion
4. Create a dbt project which will create the BigQuery view from the BigQuery table.
```

5. Create SQLMeshCLI task that will run the dbt project.

> 

Reviewer: ```suggestion
```

> ### How will we implement dbt project?

Reviewer: ```suggestion
```

SQLMesh supports integration with a variety of tools like Airflow, dbt, dlt, etc. One of the common use-cases of SQLMesh is to run dbt projects.

> You can choose to pull your dbt project from Git repository as mentioned in the [how-to-guide on dbt](docs/how-to-guides/dbt) or create the namespace files for the complete project. Here, we will create the complete project using namespace files which we will create on the fly. You can later choose to push all the namespace files to GitHub repository using [PushNamespaceFiles](docs/how-to-guides/pushnamespacefiles).

Reviewer: These links need to be relative going forward so they work in the Kestra in-app docs
```suggestion
You can choose to pull your dbt project from Git repository as mentioned in the [how-to-guide on dbt](docs/how-to-guides/dbt) or create the namespace files for the complete project. Here, we will create the complete project using namespace files which we will create on the fly. You can later choose to push all the namespace files to GitHub repository using [PushNamespaceFiles](docs/how-to-guides/pushnamespacefiles).
```

```suggestion
You can choose to pull your dbt project from Git repository as mentioned in the [How-to guide on dbt](./dbt.md) or create the namespace files for the complete project. Here, we will create the complete project using namespace files which we will create on the fly. You can later choose to push all the namespace files to GitHub repository using [PushNamespaceFiles](./pushnamespacefiles.md).
```

### Creating Flow with SQLMeshCLI task

> 

Reviewer: ```suggestion
```

> We will first create the Kestra flow.

Reviewer: ```suggestion
```

> Based on the steps mentioned in the description, lets create tasks for each step.

Reviewer: ```suggestion
Based on the steps mentioned in the description, let's create tasks for each step.
```

        fields:
        - name: order_id
          type: INT64
        - name: customer_name
          type: STRING
        - name: customer_email
          type: STRING
        - name: product_id
          type: INT64
        - name: price
          type: FLOAT64
        - name: quantity
          type: INT64
        - name: total
          type: FLOAT64

  - id: load_orders_table
    type: io.kestra.plugin.gcp.bigquery.Load
    description: Load orders table with data from orders.csv
    from: "{{ outputs.orders_http_download.uri }}"
    projectId: <gcp-project-id>
    serviceAccount: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
    destinationTable: "<gcp-project-id>.ecommerce.orders"
    format: CSV
    csvOptions:
      fieldDelimiter: ","
      skipLeadingRows: 1

  - id: sqlmesh_transform
    type: io.kestra.plugin.sqlmesh.cli.SQLMeshCLI
    description: Use SQLMesh to run the dbt project
    inputFiles:
      sa.json: "{{ secret('GCP_SERVICE_ACCOUNT_JSON') }}"
    namespaceFiles:
      enabled : true
    beforeCommands:
      - pip install "sqlmesh[bigquery]"
      - pip install dbt-bigquery
    commands:
      - sqlmesh init -t dbt
      - sqlmesh plan --auto-apply
```

> Ensure you set the `namespaceFiles` property with `enabled: true` to ensure that the task has access to namespaceFiles. Provide the GCP service account JSON file so that the task is able to connect to GCP account in order to access BigQuery. This file is referenced in the dbt project file.

Reviewer: ```suggestion
Ensure you set the `namespaceFiles` property with `enabled: true` set to ensure that the task has access to namespace files. Also, provide the GCP service account JSON file so that the task is able to connect to your GCP account in order to access BigQuery. This file is referenced in the dbt project file.
```
